src/Core/Util/Number/NumberExtractor.php

```diff
@@ -98,6 +114,35 @@ public function extract($resource, string $propertyPath): Number
         return $this->toNumber($plainValue);
     }
 
+    /**
+     * Check if object contains provided public property and extract it as a Number, else return null
+     *
+     * @param $resource
+     * @param string $property
+     *
+     * @return Number|null
+     *
+     * @throws ReflectionException
+     */
+    private function extractPublicPropertyFirst($resource, string $property): ?Number
+    {
+        if (!is_object($resource)) {
+            return null;
+        }
+
+        if (!property_exists($resource, $property)) {
+            return null;
+        }
+
+        $reflectedObj = new ReflectionClass($resource);
```

Comment:
Please a local cache system here to avoid creating too many `ReflectionClass`
Just a class field with array mapping via the classname (using `get_class`)